File: C++/Simple_Abstract_DS.cpp
template <typename T>
struct SLL{
    T val;
    SLL* next;
};

template <typename T>
struct DLL{
    T val;
    DLL* next, prev;
};

template <typename T>
struct BinaryTree{
    T val;
    BinaryTree* left;
    BinaryTree* right;
};

template <typename T>
struct Tree{
    T val;
    vector<Tree*> children;
};

File: Java/Simple_Abstract_DS.java
import java.util.*;

class TreeNode{
    int value;
    ArrayList<TreeNode> children;

    TreeNode(int v){
        value = v;
        children = new ArrayList<>();
    }
}

class BTreeNode{
    int value;
    BTreeNode left;
    BTreeNode right;

    BTreeNode(int v){
        value = v;
        left = right = null;
    }
}

class SLL{
    int value;
    SLL next;

    SLL(int v){
        value = v;
        next = null;
    }
}

class DLL{
    int value;
    DLL prev;
    DLL next;

    DLL(int v){
        value = v;
        prev = next = null;
    }
}